app/api/odm/models.ts

```diff
@@ -8,12 +8,16 @@ export type WithId<T> = T & {
   _id: Schema.Types.ObjectId;
 };
 
-export type UwaziFilterQuery<T> = FilterQuery<T> & { _id?: any };
+export type DataModelType<T> = WithId<T> & Document;
```

Comment:
this DataModelType is probably not correct, we are using this on UwaziFilterQuery and in that case the _id can be a string, this will not work in that case, can you check if this is true ?, options:
- actually enforce the ObjectId type for _id on the whole application (this really has advantages, but is probably a longer and more difficult development)
- use a diferent WithId for the UpdateQuery with supports strings as the _id

Reply:
another issue with this type, there is another Data Type defined with the "old" definition in model.ts
https://github.com/huridocs/uwazi/blob/eec9203126096f24731f543f2bea8b33bb121541/app/api/odm/model.ts#L9 and its being used already in many places (meaning we are using now 2 different data types), we should remove the new one created and use the old one ? also move all this types to model.ts where i think make more sense ?

Reply:
As for your first question, it apparently takes the string without any issue:
![MongooseTypescript](https://user-images.githubusercontent.com/4309019/135507259-0d1c33ee-d4f5-4f44-87af-00777aa2a758.jpg)

Reply:
As for your second question, I am not really sure we want the DataType to always be a Partial?  The ReadOnly is probably more secure, but I'm not sure the implications.